api/v1alpha1/policy_helpers.go

```diff
@@ -37,7 +38,15 @@ type TargetSelector struct {
 	Kind gwapiv1.Kind `json:"kind"`
 
 	// MatchLabels are the set of label selectors for identifying the targeted resource
-	MatchLabels map[string]string `json:"matchLabels"`
+	// +optional
```

Comment:
Note: empty `MatchLabels` and and empty `MatchExpressions` will match everything

Reply:
Was the previous behavior with MatchLabels? if not, let's add a release note for a breaking change. 

Also, in general, let's document these semantics in the top-level struct. 

What's the behavior if both are set? If this is not allowed, maybe we should:
- Enforce it with CEL. 
- Move to a new GW-API-like structure where we have a Match type and then the match value

```yaml

type: LabelSelector
matchLabels:
  - foo: bar
  - bar:foo 

```

Reply:
> Was this previous behavior with MatchLabels? if not, let's add a release note for a breaking change.

Yeah - nil/empty set matches everything - see [ref](https://github.com/kubernetes/apimachinery/blob/59e9003f02d6f0c8fff53719a7a0604ec82ee9a9/pkg/labels/selector.go#L931C65-L932C49)


> What's the behavior if both are set? If this is not allowed, maybe we should:

They can be used in conjunction with each other - like a regular metav1.LabelSelector. This is similar behaviour to the namespaceSelector on various k8s types.

You can see it in the parsing logic here - https://github.com/kubernetes/apimachinery/blob/a19f1f813776bd4a90401b63a96e07ed847f88d0/pkg/apis/meta/v1/helpers.go#L36